packages/stratimikos/src/Stratimikos_LinearSolverBuilder_def.hpp

```diff
@@ -541,6 +541,15 @@ void LinearSolverBuilder<Scalar>::initializeDefaults()
     );
 #endif
 
+#ifdef HAVE_STRATIMIKOS_IFPACK2
+  setPreconditioningStrategyFactory(
+    abstractFactoryStd<Thyra::PreconditionerFactoryBase<Scalar>,
+    Thyra::Ifpack2PreconditionerFactory<Tpetra::CrsMatrix<Scalar>>>(),
+    "Ifpack2", true
+    );
+#endif
```

Comment:
The updated code should allow for multiple registrations of the same preconditioner strategy without any errors.  (The latter registrations are ignored.)

Reply:
What updated code?

Reply:
https://github.com/trilinos/Trilinos/commit/f126ecf6df804c54f8a1b266ab8f148119cc1964

Reply:
Hm. That's odd. Cause there were crashes that got fixed by removing a duplicate registration: 
https://testing.sandia.gov/cdash/test/162173350

Reply:
Hum, that is not right.  How can I reproduce the problem you were seeing?

Reply:
The test fails in a couple of builds but I'm not quite sure what the common factor is:
https://testing.sandia.gov/cdash/testSummary.php?project=1&name=Stratimikos_Galeri_xpetra_double_Jacobi_Tpetra_MPI_4&date=2023-06-28
Let me debug this some more, maybe I missed something.

Reply:
@bartlettroscoe Have a look at f5197bf1c0a84d84c3804fa4277bf6b281ff790d. I think that was the issue.